File: Golang/ImplementStrStr/ImplementStrStr_test.go
package ImplementStrStr

import "testing"

func TestStrStr(t *testing.T) {

	if strStr("aaab", "b") == 3 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if strStr("", "") == 0 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if strStr("", "a") == -1 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if strStr("whats up", "s ") == 4 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/ImplementStrStrTest.py
from unittest import TestCase

from ImplementStrStr import ImplementStrStr


class TestImplementStrStr(TestCase):
    def test_strStr(self):
        iss = ImplementStrStr()

        self.assertTrue(iss.strStr("aaab", "b") == 3)

        self.assertTrue(iss.strStr("", "") == 0)

        self.assertTrue(iss.strStr("", "a") == -1)

        self.assertTrue(iss.strStr("whats up", "s ") == 4)
